src/source/image_source.js

```diff
@@ -111,6 +128,42 @@ class ImageSource extends Evented implements Source {
             }
         });
     }
+    
+    /**
+     * Updates the image URL and, optionally, the coordinates. To avoid having the image flash after changing,
+     * set the `raster-fade-duration` paint property on the raster layer to 0.
+     *
+     * @param {Object} options
+     * @param {string} [options.url] Required image URL.
+     * @param {Array<Array<number>>} [options.coordinates] Four geographical coordinates,
+     *   represented as arrays of longitude and latitude numbers, which define the corners of the image.
+     *   The coordinates start at the top left corner of the image and proceed in clockwise order.
+     *   They do not have to represent a rectangle.
+     * @returns {ImageSource} this
+     */
+    updateImage(options) {
+        if (!this.image || !options.url) {
+            return this;
+        }
+
+        var updateCoords = Boolean(options.coordinates);
+
+        getImage(this.map._transformRequest(options.url, ResourceType.Image), (err, image) => {
+            if (err) {
+                this.fire(new ErrorEvent(err));
+            } else if (image) {
+                this.image = browser.getImageData(image);
+                this.texture = null;
+                this.fire(new Event('data', {dataType:'source', sourceDataType: 'content'}));
+            }
+
+            if (updateCoords && this.map) {
```

Comment:
Just check `options.coordinates` here instead of casting to a Boolean and saving the result as a variable.